src/query/src/optimizer/scan_hint.rs

```diff
@@ -121,6 +121,30 @@ impl ScanHintRule {
             });
         }
         adapter.with_ordering_hint(&opts);
+
+        let mut sort_expr_cursor = order_expr.iter().map(|s| s.expr.try_as_col()).flatten();
+        let region_metadata = adapter.region_metadata();
+        let mut pk_column_iter = region_metadata.primary_key_columns();
+        let mut curr_sort_expr = sort_expr_cursor.next();
+        let mut curr_pk_col = pk_column_iter.next();
+
+        while let (Some(sort_expr), Some(pk_col)) = (curr_sort_expr, curr_pk_col) {
+            if sort_expr.name == pk_col.column_schema.name {
+                curr_sort_expr = sort_expr_cursor.next();
+            }
+            curr_pk_col = pk_column_iter.next();
+        }
+
+        let next_remaining = sort_expr_cursor.next();
+        match (curr_sort_expr, next_remaining) {
+            (Some(expr), None)
+                if expr.name == region_metadata.time_index_column().column_schema.name =>
+            {
+                adapter.with_distribution(TimeSeriesDistribution::PerSeries);
```

Comment:
Do we need to modify the `parallelize_scan`? Now, it assigns partition ranges without considering the time series distribution. We may output the same time series in different partitions.

Reply:
Now it'll skip parallelizing for time series distribution